### DISCUSSION TITLE
API to get access to Topology

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1372] [Q&A] [answered]
As part of zigpy Application.py and toplogy.py I found APIs to start stop topology reports, but do not find any to retreive one instance of this.

Does that mean that we should extract the information directly from the DB, or does those APIs must be developped ?

I guess in ZHA there is some way to a graphical representation of the Zigbee network. What are they using to get this informations ?

**Comment by puddly** [comment, marked as answer]
Everything is stored under `app.topology.neighbors` and `app.topology.routes`: https://github.com/zigpy/zigpy/blob/3d1b60c9181fb40638eaf3e70c5815967b07300c/zigpy/topology.py#L52-L55

**Reply by pipiche38** [reply]
No API, just acceding to the data.

**Comment by pipiche38** [comment]
is there a way to know when a full cycle is done ? In other words, when the message "Finished scanning neighbors for all devices" ?
